src/app/state/account/saga.ts

```diff
@@ -17,6 +17,17 @@ import { selectAccountAddress } from './selectors'
 export function* fetchAccount(action: PayloadAction<string>) {
   const address = action.payload
 
+  if (!isValidAddress(address)) {
+    yield* put(
+      actions.accountError({
```

Comment:
Account error is set in app via this action, but AccountPage is not selecting accountError from a store. It has its own check.

Reply:
That's true, but my intention was to keep the app's internal data model (which is independent of the UI) consistent.
If we don't register an error here, the state stored in redux will look like as if we have a valid account loaded, which is definitely not the case. I wanted to avoid that, even if the UI is not visible because of the check on AccountPage.

But if you say that we don't have to care about the consistency of the internal app state  (since the UI is hidden anyway), I'm happy to remove it.

Reply:
I have nothing against keeping an error in state. I was more curious why we are not using it in AccountPage to render AlertBox.

Reply:
Ah, I see. There is actually a reason for that: the `AccountPage` is rendered _before_ the account loading saga is triggered. So if we used the presence of the error as the condition, then when initializing the app, there would be a short while when the error is not yet marked in the state, and we try to render the UI, and get an exception at `PrerttyAddress`, which then crashed the whole UI, and it doesn't recover any more.

To avoid this race condition, it's easier and safer to simply check the validity of the address in the UI, too.